processing/src/main/java/org/apache/druid/java/util/common/guava/Yielders.java

```diff
@@ -80,4 +81,27 @@ public void close() throws IOException
       }
     };
   }
+  public static <T> void safeCloseYielder(Throwable t1, Yielder<T> yielder)
+  {
+    try {
+      if (yielder != null) {
+        yielder.close();
+      }
+    }
+    catch (Throwable t2) {
+      t1.addSuppressed(t2);
+    }
+  }
+  public static <T> Yielder<T> withClose(Sequence<T> in, Function<Sequence<T>, Yielder<T>> fn)
+  {
+    Yielder<T> yielder = null;
+    try {
+      yielder = fn.apply(in);
+    }
+    catch (Throwable t1) {
+      safeCloseYielder(t1, yielder);
+      throw t1;
+    }
+    return yielder;
+  }
```

Comment:
How can `yielder` be anything but `null` if `fn.apply` throws an exception?  The catch statement here is basically not doing anything...